src/background.js

```diff
@@ -28,6 +33,32 @@ browser.browserAction.onClicked.addListener(() => {
     openOverview()
 })
 
+
+browser.bookmarks.onCreated.addListener((id, bookmarkInfo) => {
+    //TO-DO Add method to handle indexing of bookmark
+    //bookmarkInfo.url gives the URL of the bookmark
+    //id is the unique id of the bookmark
+    //https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/bookmarks/onCreated
+    //Check if there already exists a page with this URL.
+    const samePageCandidates = (await findPagesByUrl({url})).rows.map(row => row.doc);
+    const reusablePage = await tryReidentifyPage({null, bookmarkInfo.url, samePageCandidates})
```

Comment:
`tryReidentifyPage`, while still not implemented, is intended for the flow of a page visit, as it requires a browser tab. I think easier here is to just set `reusablePage` to the latest page from `samePageCandidates` results, if any. If `reusablePage` not null, it should skip the `fetchPageData()` and related logic below.